extensions/blocks/instagram-gallery/image-transition.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState, useRef } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+
+export default function ImageTransition( { src } ) {
+	const [ loaded, setLoaded ] = useState( false );
+	const [ size, setSize ] = useState( 'auto' );
+
+	const img = useRef();
+
+	const transitionSpeed = '.5';
+
+	const onImageLoad = () => {
+		setLoaded( true );
+		setSize( 'auto' );
+	};
+
+	useEffect( () => {
+		const imgSrc = src;
+		if ( imgSrc ) {
+			// First pre-load image as a new window.Image and then update local state when it is loaded,
+			// this lets us handle the fade in animation
+			const tmpImg = new window.Image();
+			tmpImg.src = imgSrc;
+			tmpImg.onload = onImageLoad;
+		}
+		if ( img.current ) {
+			setSize( img.current.parentNode.getBoundingClientRect().width );
+		}
+	}, [] );
```

Comment:
Missing the `src` dependency.

Reply:
This is deliberate, we only want this effect to run once on component mount, and having an empty dependency array ensures this.

Reply:
... although I think the need to only run it on mount was related to that funny isMounted hack in the first iteration - works fine with src in the dependencies now, and that keeps the linter happier to so have added it in.